pkg/network/ebpf/c/tracer-stats.h

```diff
@@ -92,16 +94,11 @@ static __always_inline void update_conn_stats(conn_tuple_t *t, size_t sent_bytes
 }
 
 static __always_inline void update_tcp_stats(conn_tuple_t *t, tcp_stats_t stats) {
-    // query stats without the PID from the tuple
-    __u32 pid = t->pid;
-    t->pid = 0;
-
     // initialize-if-no-exist the connetion state, and load it
     tcp_stats_t empty = {};
-    bpf_map_update_with_telemetry(tcp_stats, t, &empty, BPF_NOEXIST);
+    bpf_map_update_with_telemetry(tcp_stats, &(t->cookie), &empty, BPF_NOEXIST);
```

Comment:
Can we change the update pattern here, to first check for existence via lookup and then update? The telemetry of this map indicates a huge number of failures of the update operation with the error "file exists". Essentially, for the duration of the connection the map_update is successful once and then keeps failing. We can minimize the unnecessary helper calls by changing the order of lookup and update.
What do you think?

Reply:
We would have to do something like lookup -> update -> lookup (in case update fails). We need the BPF_NOEXIST flag to have only one update call win for the same tuple. It should reduce the calls to update, but the pattern would be a little confusing; I can put in a comment there.